index/secure_index.go

```diff
@@ -0,0 +1,64 @@
+package index
+
+import (
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/binary"
+	"fmt"
+	"github.com/jxguan/go-datastructures/bitarray"
```

Comment:
non-standard-lib packages should go in a separate block of imports, after a blank line. Like:

```go
import (
    "crypto/sha256"
    ...
    "hash"

    "github.com/jxguan/go-datastructures/bitarray"
)
```